data-processing-lib/src/data_processing/pure_python/transform_table_processor.py

```diff
@@ -0,0 +1,190 @@
+        # Add data access and statistics to the processor parameters
+        transform_params = dict(params.get_transform_params())
+        transform_params["data_access"] = self.data_access
+        transform_params["statistics"] = statistics
+        # Create local processor
+        self.transform = params.get_transform_class()(transform_params)
+        # Create statistics
+        self.stats = statistics
+        self.last_fname = None
+        self.last_fname_next_index = None
+
+    def process_data(self, f_name: str) -> None:
+        """
+        Method processing an individual file
+        :param f_name: file name
+        :return: None
+        """
+        logger.debug(f"Begin processing file {f_name}")
+        if self.data_access is None:
+            logger.warning("No data_access found. Returning.")
+            return
+        t_start = time.time()
+        # Read source table
+        table = self.data_access.get_table(path=f_name)
+        if table is None:
+            logger.warning("File read resulted in None. Returning.")
+            self.stats.add_stats({"failed_reads": 1})
+            return
+        self.stats.add_stats({"source_files": 1, "source_size": table.nbytes})
+        # Process input table
+        try:
+            if table.num_rows > 0:
+                # execute local processing
+                logger.debug(f"Begin transforming table from {f_name}")
+                out_tables, stats = self.transform.transform(table=table)
+                logger.debug(f"Done transforming table from {f_name}")
+            else:
+                logger.info(f"table: {f_name} is empty, skipping processing")
+                self.stats.add_stats({"skipped empty tables": 1})
+                return
+            # save results
+            self._submit_table(f_name=f_name, t_start=t_start, out_tables=out_tables, stats=stats)
+        except Exception as e:
+            logger.warning(f"Exception {e} processing file {f_name}: {traceback.format_exc()}")
+            self.stats.add_stats({"transform execution exception": 1})
+
+    def flush(self) -> None:
+        """
+        This is supporting method for transformers, that implement buffering of tables, for example coalesce.
+        These transformers can have buffers containing tables that were not written to the output. Flush is
+        the hook for them to return back locally stored tables and their statistics.
+        :return: None
+        """
+        t_start = time.time()
+        if self.last_fname is None:
+            # for some reason a given worker never processed anything. Happens in testing
+            # when the amount of workers is greater then the amount of files
+            logger.debug("skipping flush, no name for file is defined")
+            return
+        try:
+            # get flush results
+            logger.debug(f"Begin flushing transform")
+            out_tables, stats = self.transform.flush()
```

Comment:
I'd put ` t_start = time.time()` before this line

Reply:
nah, its just debug